### ISSUE TITLE
Derelict Scout

### ISSUE THREAD

**Issue by Scara83** [freeorion/freeorion#468]
I had the feeling of scouting systems that I couldn't see at that point but didn't know why.
I think I found out know, it seems to be the effect of the Derelict Scout.
It increases the scouting radius, but the effect seems to be taken to the next system on the route that is 1 jump away in my case.
The increased scouting radius isn't visible any more but systems are, that weren't before.
I got the derelict scout in deep space at the mousepointer and had view half way to thalassa, so thalassa and rigel weren't visible.
Going to Zibal they suddenly got visible, clearly not being in scouting range yet. 
![derelictscout](https://cloud.githubusercontent.com/assets/17101396/12868979/3d7860a4-cd13-11e5-97a5-a3ddc6d3352c.jpg)

**Comment by geoffthemedio** [comment]
I can't tell from the description or picture what the problem is. Is it just that the increased detection range is being applied to a moving ship before it moves, but used to determine what can be seen after it moves? If so, that's just how it works, or a flaw in the design of the derelict special given the timing of effects application, movement, and visibility updates (which tend to be done often to ensure info that can be seen at some point isn't missed even if later updates or turn processing steps would make it so that same info can't be seen).

**Comment by Vezzra** [comment]
If I understand correctly, the problem seems to be that the scout found a Derelict special at the empty system next to the mouse pointer, which granted extended vision, but apparently didn't include the Thalassa and Rigel systems.

The scout proceeded to Zibal the next turn (I guess that jump had been made within just one turn), and suddenly also Thalassa and Rigel became visible, although clearly outside the normal detection range of the scout. If that is what happened, apparently the extended vision from the Derelict special was still applied when the scout arrived at Zibal, which is probably not how things are intended.

Might require a rewrite of the Derelict special...

**Comment by MatGB** [comment]
It's been known for ages, Dilvish and I did some work on it ages back to make sure that the timings were as good as we could get it (it used to be that if you moved to a new system that also had a derelict you lost the vision bonus from that one as well when the first wore off).

If a scout (or any ship) with the bonus moves to a new system in exactly one turn then it will get a flash of visibility of the radius from the new system.

The only way I can see to fix it purely in FOCS would be to have the bonus apply as a high priority effect and then be removed as a low priority effect on the same turn, so  you get the map info but not actual visibility of the systems (I think). Not sure that's an optimal solution and not sure it's a major problem as is, but if someone wants to rewrite either the special or the visibility processing backend well, I'll happily test changes.

**Comment by geoffthemedio** [comment]
I've been working on a SetVisibility effect that could replace the detection meter adjustment currently used.

**Comment by Vezzra** [comment]
A SetVisibility effect would certainly make scripting the Derelict special (and similar things) easier and clearer.

**Comment by geoffthemedio** [comment]
Done, seems to work in an initial test.
https://github.com/freeorion/freeorion/commit/2e5d5f79c2e0bc70caec2cb9ea7cc267b33a98d3

```
EffectsGroup
    scope = Planet
    effects = SetVisibility visibility = Full empire = Source.Owner
```

Effect-set visibility is applied between a few other steps in the next visibility update after the effects are executed.

**Comment by Vezzra** [comment]
Nice! @MatGB, you think you can squeeze in a little rewrite of the Derelict special to make use of the new effect? Not urgent, just so we don't forget it to do at some point and can close this issue.

**Comment by Vezzra** [comment]
Ok, should be fixed by #619, closing this issue.
